### ISSUE TITLE
Do translation macro really need to be readOnly?

### ISSUE THREAD

**Issue by nag5000** [ember-intl/ember-intl#1166]
- [x] I am on the [latest](https://github.com/ember-intl/ember-intl/releases/latest) ember-intl version
- [x] I have searched the [issues](https://github.com/ember-intl/ember-intl/issues) of this repo and believe that this is not a duplicate

### Environment

- Ember Version: 3.4.8
- Ember Intl Version: 4.2.2

Hello! I'm migrating from ember-i18n to ember-intl, and I had a question:

Do translation macro *really* need to be [readOnly](https://github.com/ember-intl/ember-intl/blob/c6ab6c1523e011381b985d1281b7d8ea0eb80902/addon/macros/intl.js#L23)?

Sometimes in my app I use translation macro as a default value in a component that can be overridden via template. Like this:
```
// components/foo-bar.js
export default Component.extend({
  btnText: t('translation.key');
});

// templates/qux.hbs
{{foo-bar btnText=(t "another.translation.key")}}
```

It worked with ember-i18n. But it doesn't work with ember-intl, because the macro has `readOnly()` modifier. Is it really needed, regardless of how bad my usage pattern is? I understand that the problem can be easily solved using an additional computed property or helper.

**Comment by jasonmit** [comment]
@nag5000 that's a fair point and something we should support.

@buschtoens are you fine with removing the readOnly here?

**Comment by jasonmit** [comment]
Should be fixed in 4.3.0
